app/models/king.rb

```diff
@@ -1,2 +1,11 @@
 class King < Piece
+  def valid_move?(destination_x, destination_y)
+    if destination_x.between?(0, 7) && destination_y.between?(0, 7)
+      difference_x = destination_x - position_x
+      difference_y = destination_y - position_y
+
+      return true if difference_x.abs == 1 || difference_y.abs == 1
+    end
+    false
+  end
 end
```

Comment:
Well done! Looks great.

**Ways to improve it:**

1. Use a  "guard statement". This lets you exit a method early if it doesn't satisfy certain conditions.
You can put this on the first line of the method.

`return false unless destination_x.between?(0, 7) && destination_y.between?(0, 7)`

Then you no longer need the `if` statement. Or the nesting. Makes the method very easy to read.

2. I believe this line has an error

`difference_x.abs == 1 || difference_y.abs == 1`

If the y difference is 5. This will still return true. Because the `||` (`or`) returns true if **either** statement is true.

Try this instead.

`VALID_MOVE_DISTANCES = [0,1]`

`VALID_MOVE_DISTANCES.include?(difference_x.abs) && VALID_MOVE_DISTANCES.include?(difference_y.abs)`

^ This can be the last line of your method because ruby will evaluate it and return the value.

Reply:
Thanks, however I made these changes, and now one test is failing that was passing before: it `returns false if the move is the same space`. Before this refactoring, it returned false because the `difference_x` and `difference_y` were both `0`, meaning they would not pass the following line:

`return true if difference_x.abs == 1 || difference_y.abs == 1`

Please advise. Thanks.